### ISSUE TITLE
Load table from csv file: response error when works fine

### ISSUE THREAD

**Issue by JSAURAJ** [googleapis/nodejs-bigquery#329]
#### Environment details

  - OS: Ubuntu 18.04 LTS
  - Node.js version: 8.14.0
  - npm version: 6.4.1
  - `@google-cloud/bigquery` version:@google-cloud/bigquery@1.3.0

#### Steps to reproduce

We start from a CSV that has 2 incorrect lines and we want to load in a bigquery table.

1. When maxBadRecords=0, the job returns an error and the data is not inserted into the table. Expected response. Ok

2. When maxBadRecords=(>=2), the job returns the same error, but the data in the destination is inserted correctly. It is an unexpected response according to the official documentation

[Documentation ](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-csv#bigquery_load_table_gcs_csv-nodejs) says:
maxBadRecords | (Optional) The maximum number of bad records that BigQuery can ignore when running the job. If the number of bad records exceeds this value, an invalid error is returned in the job result. The default value is 0, which requires that all records are valid

Thanks

**Comment by callmehiphop** [comment]
It looks like this issue might have been resolved in an update to our core library (yay). However, while trying to troubleshoot I found a new issue (noo) with our job polling code. As it is today, it looks like we try and throw an exception if the `status.errors` is present via `jobs.get`. However I think this logic might be wrong, it seems like instead of checking against `status.errors`, we actually want to use `status.errorResult` to determine if we want to throw an exception or ignore `status.errors`.

@tswast does that sound right?
